src/dump.c

```diff
@@ -37,29 +38,43 @@ const char * dump_string_get_tag_type(flv_tag * tag) {
 }
 
 const char * dump_string_get_video_codec(flv_video_tag tag) {
-    switch (flv_video_tag_codec_id(tag)) {
-        case FLV_VIDEO_TAG_CODEC_JPEG: return "JPEG";
-        case FLV_VIDEO_TAG_CODEC_SORENSEN_H263: return "Sorenson H.263";
-        case FLV_VIDEO_TAG_CODEC_SCREEN_VIDEO: return "Screen video";
-        case FLV_VIDEO_TAG_CODEC_ON2_VP6: return "On2 VP6";
-        case FLV_VIDEO_TAG_CODEC_ON2_VP6_ALPHA: return "On2 VP6 with alpha channel";
-        case FLV_VIDEO_TAG_CODEC_SCREEN_VIDEO_V2: return "Screen video version 2";
-        case FLV_VIDEO_TAG_CODEC_AVC: return "AVC";
-        default: return "Unknown";
+    if (flv_video_tag_is_ext_header(tag.video_tag))
+    {
+        if (tag.fourcc == FLV_VIDEO_FOURCC_HEVC)
+            return "HEVC";
+        else if (tag.fourcc == FLV_VIDEO_FOURCC_AV1)
+            return "AV1";
+        else if (tag.fourcc == FLV_VIDEO_TAG_CODEC_VP9)
+            return "VP9";
+        else
+            return "Unknown";
```

Comment:
Oh I see, I hadn't realized there were both
```C
/* Extended RTMP codecs */
#define FLV_VIDEO_TAG_CODEC_AV1             99
#define FLV_VIDEO_TAG_CODEC_VP9             100
#define FLV_VIDEO_TAG_CODEC_HEVC            101
```
and
 ```C
#define FLV_VIDEO_FOURCC_AV1        FOURCC('a', 'v', '0', '1')
#define FLV_VIDEO_FOURCC_VP9        FOURCC('v', 'p', '0', '9')
#define FLV_VIDEO_FOURCC_HEVC       FOURCC('h', 'v', 'c', '1')
```

Given that the latter are now defined as integers, you probably want to use them instead of the former, since their value seems to be arbitrary.

In the end, I think the codec list could be:

```C
/* video tag */
#define FLV_VIDEO_TAG_CODEC_JPEG            1
#define FLV_VIDEO_TAG_CODEC_SORENSEN_H263   2
#define FLV_VIDEO_TAG_CODEC_SCREEN_VIDEO    3
#define FLV_VIDEO_TAG_CODEC_ON2_VP6         4
#define FLV_VIDEO_TAG_CODEC_ON2_VP6_ALPHA   5
#define FLV_VIDEO_TAG_CODEC_SCREEN_VIDEO_V2 6
#define FLV_VIDEO_TAG_CODEC_AVC             7

/* Extended RTMP codecs */
#define FLV_VIDEO_FOURCC_AV1        FOURCC('a', 'v', '0', '1')
#define FLV_VIDEO_FOURCC_VP9        FOURCC('v', 'p', '0', '9')
#define FLV_VIDEO_FOURCC_HEVC       FOURCC('h', 'v', 'c', '1')
```

so there's no confusion.

Reply:
Ah gotcha, good catch. I changed.